easy-ui-react/src/IconButton/IconButton.tsx

```diff
@@ -0,0 +1,51 @@
+import React, { useRef, ReactElement } from "react";
+import { useButton } from "react-aria";
+import { IconProps } from "../Icon";
+import { ButtonColor } from "../Button";
+import { classNames, variationName } from "../utilities/css";
+import { logWarningIfInvalidColorVariantCombination } from "../Button/utilities";
+import styles from "./IconButton.module.scss";
+import commonButtonStyles from "../Button/Button.module.scss";
+
+export type IconButtonVariant = "filled" | "outlined";
+
+export type IconButtonProps = {
+  /** Button color */
+  color?: ButtonColor;
+  /** Button variant */
+  variant?: IconButtonVariant;
+  /** Icon */
+  icon: ReactElement<IconProps>;
+  /** Disables button */
+  isDisabled?: boolean;
+};
+
+export function IconButton(props: IconButtonProps) {
+  const {
+    color = "primary",
+    variant = "filled",
+    icon,
+    isDisabled = false,
+  } = props;
+
+  const ref = useRef(null);
+  const { buttonProps } = useButton(props, ref);
+
+  logWarningIfInvalidColorVariantCombination(color, variant);
+
+  return (
+    <button
+      disabled={isDisabled}
+      ref={ref}
+      className={classNames(
+        styles.IconButton,
+        styles[variationName("variant", variant)],
+        commonButtonStyles[variationName("color", color)],
+        commonButtonStyles[variationName("variant", variant)],
+      )}
+      {...buttonProps}
+    >
+      {React.cloneElement(icon)}
```

Comment:
**issue:** i didn't catch this pattern in the original `Button` pull request, but i believe it would be better to accept a symbol as the prop instead of cloning an `<Icon />` component.

in addition, i would suggest using an accessibility pattern directly within the `<button />` instead of using the underlying `<Icon accessibilityLabel="" />` due to it being marginally more robust across screen readers: https://www.sarasoueidan.com/blog/accessible-icon-buttons/#there-is-no-one-way-to-rule-them-all:~:text=Personally%2C%20I%E2%80%99d%20not,on%20his%20findings.

> Personally, I’d not use the SVG icon itself to provide a label for the button when I can provide one on the button itself directly. As Scott states, there’s just no good way to use an SVG as the sole means to provide an accessible name to a link or button. Every test had failures. Thankfully, Scott has saved us hours of testing, so if you need to provide a label to the button using the svg itsef, you can choose the pattern you need based on his findings.

example of changes:

```tsx
import type { IconSymbol } from "../types";

export type IconButtonProps = {
  /** Icon symbol */
  symbol: IconSymbol;
  /** Description of icon */
  accessibilityLabel: string;
};

export function IconButton(props: IconButtonProps) {
  const {
    symbol,
    accessibilityLabel,
  } = props;
  return (
    <button {...buttonProps}>
      <Text visuallyHidden>{accessibilityLabel}</Text>
      <Icon symbol={symbol} />
    </button>
  );
}

// then use like
<IconButton
  symbol={BackIcon}
  accessibilityLabel="Back"
/>
```

with this paradigm, it allows the `IconButton` component to control the rendering of the icon. with cloning, the user has too much flexibility in changing the rendering of the icon within the button to the point where it would be easy to break. for example, the user could break the size and color scheme with the following: `<IconButton icon={<Icon size="lg" color="blue-800" />} />`

i would also highly suggest the same pattern with the `icon*` props in the original `Button` component for the same reasons, but that could be a follow-up task

Reply:
very interesting, thanks for the info! updated `IconButton` and `Button` . for `Button` i omitted the `Text` component since i imagine the accessible text is given by the `Button` text itself